tests/unit/scheduling.test.ts

```diff
@@ -27,81 +52,115 @@ test("Test reviewing with default settings", () => {
 test("Test reviewing with default settings & delay", () => {
     const delay = 2 * 24 * 3600 * 1000; // two day delay
     expect(
-        schedule(ReviewResponse.Easy, 10, DEFAULT_SETTINGS.baseEase, delay, DEFAULT_SETTINGS, {}),
+        osrSchedule(
+            ReviewResponse.Easy,
+            10,
+            DEFAULT_SETTINGS.baseEase,
+            delay,
+            DEFAULT_SETTINGS,
+            emptyHistogram,
+        ),
     ).toEqual({
         ease: DEFAULT_SETTINGS.baseEase + 20,
         interval: 42,
     });
 
     expect(
-        schedule(ReviewResponse.Good, 10, DEFAULT_SETTINGS.baseEase, delay, DEFAULT_SETTINGS, {}),
+        osrSchedule(
+            ReviewResponse.Good,
+            10,
+            DEFAULT_SETTINGS.baseEase,
+            delay,
+            DEFAULT_SETTINGS,
+            emptyHistogram,
+        ),
     ).toEqual({
         ease: DEFAULT_SETTINGS.baseEase,
         interval: 28,
     });
 
     expect(
-        schedule(ReviewResponse.Hard, 10, DEFAULT_SETTINGS.baseEase, delay, DEFAULT_SETTINGS, {}),
+        osrSchedule(
+            ReviewResponse.Hard,
+            10,
+            DEFAULT_SETTINGS.baseEase,
+            delay,
+            DEFAULT_SETTINGS,
+            emptyHistogram,
+        ),
     ).toEqual({
         ease: DEFAULT_SETTINGS.baseEase - 20,
         interval: 5,
     });
 });
 
 test("Test load balancing, small interval (load balancing disabled)", () => {
-    const dueDates = {
+    const dueDates = new DueDateHistogram({
         0: 1,
         1: 1,
         2: 1,
         3: 4,
-    };
+    });
     expect(
-        schedule(ReviewResponse.Good, 1, DEFAULT_SETTINGS.baseEase, 0, DEFAULT_SETTINGS, dueDates),
+        osrSchedule(
+            ReviewResponse.Good,
+            1,
+            DEFAULT_SETTINGS.baseEase,
+            0,
+            DEFAULT_SETTINGS,
+            dueDates,
+        ),
     ).toEqual({
         ease: DEFAULT_SETTINGS.baseEase,
         interval: 3,
     });
-    expect(dueDates).toEqual({
+    /* expect(dueDates).toEqual(new DueDateHistogram({
         0: 1,
         1: 1,
         2: 1,
         3: 5,
-    });
+    })); */
```

Comment:
Has the scheduling behavior changed? I've tried this locally and the commented assertion is now failing. Same for the other assertions that have been commented out in `tests/unit/scheduling.test.ts`.

Reply:
Definitely the intention has been to keep scheduling (all functionality really) identical to the current behaviour. I'm not sure what happened here, but I will look into it.

Reply:
I had modified `osrSchedule` to treat `dueDateHistogram` as an input only (and to not modify it as well). I commented out the test case temporarily, and forgot to fix afterwards. Corrected now.

Reply:
@ronzulu, I'll try see what has changed as well. For now this is good to go in. There's a merge conflict though